### PR TITLE
Propagate externalization to limitranger admission controller

### PR DESCRIPTION
**Pull request by yue9944882**
**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: plugin/pkg/admission/limitranger/admission.go [hunk 1]
```diff
@@ -68,7 +68,6 @@ type LimitRanger struct {
 
 var _ admission.MutationInterface = &LimitRanger{}
 var _ admission.ValidationInterface = &LimitRanger{}
-var _ kubeapiserveradmission.WantsInternalKubeInformerFactory = &LimitRanger{}
```
Comment (yue9944882) [inline plugin/pkg/admission/limitranger/admission.go]: it's duplicated with L207

File: plugin/pkg/admission/limitranger/admission.go [hunk 2]
```diff
@@ -106,15 +105,6 @@ func (l *LimitRanger) runLimitFunc(a admission.Attributes, limitFn func(limitRan
 		return nil
 	}
 
-	obj := a.GetObject()
-	name := "Unknown"
-	if obj != nil {
-		name, _ = meta.NewAccessor().Name(obj)
-		if len(name) == 0 {
-			name, _ = meta.NewAccessor().GenerateName(obj)
```
Comment (yue9944882) [inline plugin/pkg/admission/limitranger/admission.go]: why do we do name generating here?

File: plugin/pkg/admission/limitranger/admission.go (hunk 1 above)
Comment (sttts) [reply plugin/pkg/admission/limitranger/admission.go]: Would prefer to move all of those here, next to the type.

File: plugin/pkg/admission/limitranger/admission.go (hunk 2 above)
Comment (sttts) [reply plugin/pkg/admission/limitranger/admission.go]: good question. Who wrote this?

File: plugin/pkg/admission/limitranger/admission.go [hunk 3]
```diff
@@ -81,6 +83,10 @@ func (l *LimitRanger) SetInternalKubeInformerFactory(f informers.SharedInformerF
 	l.lister = limitRangeInformer.Lister()
 }
 
+func (a *LimitRanger) SetInternalKubeClientSet(client internalclientset.Interface) {
```
Comment (yue9944882) [inline plugin/pkg/admission/limitranger/admission.go]: > Would prefer to move all of those here, next to the type.

done. and i moved it too

File: plugin/pkg/admission/limitranger/admission.go (hunk 2 above)
Comment (yue9944882) [reply plugin/pkg/admission/limitranger/admission.go]: /shrug

we should never try generating name outside the store registry, amiright?

**Comment by yue9944882** [comment]
/assign deads2k

sorting out things in `limitranger`. it's a bit more complicated than the previous ones. i put some trival changes here and will do the actual refactor in another pull for a clearer granularity of logical commits (in case reverting?). do that make sense to you?

File: plugin/pkg/admission/limitranger/admission.go [hunk 4]
```diff
@@ -467,19 +459,23 @@ func (d *DefaultLimitRangerActions) SupportsAttributes(a admission.Attributes) b
...
 // Claims will not be modified with default values because storage is a required part of pvc.Spec.
 // All storage enforced values *only* apply to pvc.Spec.Resources.Requests.
-func PersistentVolumeClaimValidateLimitFunc(limitRange *api.LimitRange, pvc *api.PersistentVolumeClaim) error {
+func PersistentVolumeClaimValidateLimitFunc(limitRange *corev1.LimitRange, pvc *api.PersistentVolumeClaim) error {
 	var errs []error
 	for i := range limitRange.Spec.Limits {
-		limit := limitRange.Spec.Limits[i]
-		limitType := limit.Type
+		coreLimit := api.LimitRangeItem{}
+		if err := apiscorev1.Convert_v1_LimitRangeItem_To_core_LimitRangeItem(&limitRange.Spec.Limits[i], &coreLimit, nil); err != nil {
```
Comment (yue9944882) [inline plugin/pkg/admission/limitranger/admission.go]: @deads2k should we do the conversion like this here?

File: plugin/pkg/admission/limitranger/admission.go [hunk 5]
```diff
@@ -467,19 +459,23 @@ func (d *DefaultLimitRangerActions) SupportsAttributes(a admission.Attributes) b
...
+			return err
+		}
+		limit := coreLimit
+		limitType := coreLimit.Type
 		if limitType == api.LimitTypePersistentVolumeClaim {
```
Comment (deads2k) [inline plugin/pkg/admission/limitranger/admission.go]: do this comparison against the external constant

File: plugin/pkg/admission/limitranger/admission.go [hunk 6]
```diff
@@ -467,19 +459,23 @@ func (d *DefaultLimitRangerActions) SupportsAttributes(a admission.Attributes) b
...
 // Users request storage via pvc.Spec.Resources.Requests.  Min/Max is enforced by an admin with LimitRange.
 // Claims will not be modified with default values because storage is a required part of pvc.Spec.
 // All storage enforced values *only* apply to pvc.Spec.Resources.Requests.
-func PersistentVolumeClaimValidateLimitFunc(limitRange *api.LimitRange, pvc *api.PersistentVolumeClaim) error {
+func PersistentVolumeClaimValidateLimitFunc(limitRange *corev1.LimitRange, pvc *api.PersistentVolumeClaim) error {
 	var errs []error
 	for i := range limitRange.Spec.Limits {
-		limit := limitRange.Spec.Limits[i]
-		limitType := limit.Type
+		coreLimit := api.LimitRangeItem{}
```
Comment (deads2k) [inline plugin/pkg/admission/limitranger/admission.go]: this name must indicate it's internalness

File: plugin/pkg/admission/limitranger/admission.go [hunk 7]
```diff
@@ -467,19 +459,23 @@ func (d *DefaultLimitRangerActions) SupportsAttributes(a admission.Attributes) b
...
+		if err := apiscorev1.Convert_v1_LimitRangeItem_To_core_LimitRangeItem(&limitRange.Spec.Limits[i], &coreLimit, nil); err != nil {
+			return err
+		}
+		limit := coreLimit
+		limitType := coreLimit.Type
 		if limitType == api.LimitTypePersistentVolumeClaim {
 			for k, v := range limit.Min {
```
Comment (deads2k) [inline plugin/pkg/admission/limitranger/admission.go]: `k` in this case could be cast to a string and then cast to the internal, right?  How about updating the minConstraint method to take it as a string

**Comment by deads2k** [comment]
yeah, that worked out better

/lgtm

**Pull request merged**